editoast/src/models/paced_train.rs

```diff
@@ -70,8 +71,23 @@ impl PacedTrain {
             train_schedule.comfort = change_group.comfort;
             train_schedule.rolling_stock_name = change_group.rolling_stock_name.clone();
         }
+
         if let Some(change_group) = &exception.rolling_stock_category {
-            train_schedule.category = change_group.value.clone();
+            match &change_group.value {
+                Some(TrainCategory::Main { main_category }) => {
+                    train_schedule.category = Some(TrainCategory::Main {
+                        main_category: *main_category,
+                    });
+                }
+                Some(TrainCategory::Sub {
+                    sub_category_code: _,
+                }) => {
+                    train_schedule.category = None;
+                }
+                None => {
+                    train_schedule.category = None;
+                }
+            }
```

Comment:
```suggestion
            train_schedule.category = change_group.value.as_main().cloned();
```

this could be made more concise